js/sagas/onboarding.js

```diff
@@ -44,8 +50,52 @@ function* pinCheckSaga(): Saga<void> {
     })
     yield put(navigateToOnboardingPinScreenAction)
 
-    // Here we wait the user to create a PIN
-    yield take(PIN_CREATE_REQUEST)
+    let isPinSaved = false
+
+    // Loop until PIN successfully saved in the Keystore
+    while (!isPinSaved) {
```

Comment:
is there any way to cancel the action ?

Reply:
If you mean cancel explictly, we could add a race [PIN_CREATE_REQUEST, PIN_CREATE_CANCEL] than if PIN_CREATE_CANCEL Win in some way we exit from the loop. The problem is that the user MUST complete the PinStep. Which use case are you thinking of?

Reply:
If the user is in the PinScreen it is already LoggedIn and must complete the Onboarding. In which case(s) we must break this Loop?